storage/series/buffer.go

```diff
@@ -351,6 +353,24 @@ func (b *dbBuffer) computedForEachBucketAsc(
 	return result
 }
 
+func (b *dbBuffer) Snapshot(ctx context.Context, blockStart time.Time) [][]xio.SegmentReader {
+	// TODO(r): pool these results arrays
+	var res [][]xio.SegmentReader
+	b.forEachBucketAsc(func(bucket *dbBufferBucket) {
+		if !bucket.canRead() {
+			return
+		}
+
+		if !blockStart.Equal(bucket.start) {
+			return
+		}
+
+		res = append(res, bucket.streams(ctx))
```

Comment:
Could you put a quick comment explaining why this is safe (i.e. bucket.streams() returns a copy...)

Reply:
done